src/runtime/signal_unix.go

```diff
@@ -769,19 +772,35 @@ func sighandler(sig uint32, info *siginfo, ctxt unsafe.Pointer, gp *g) {
 			// The faulting m is crashing first so it is the faulting thread in the core dump (see issue #63277):
 			// in expected operation, the first m will wait until the last m has received the SIGQUIT,
 			// and then run crash/exit and the process is gone.
-			// However, if it spends more than 5 seconds to send SIGQUIT to all ms,
-			// any of ms may crash/exit the process after waiting for 5 seconds.
+			// However, if it spends more than 10 seconds to send SIGQUIT to all ms,
+			// any of ms may crash/exit the process after waiting for 10 seconds.
 			print("\n-----\n\n")
 			raiseproc(_SIGQUIT)
 		}
 		if isCrashThread {
-			i := 0
-			for (crashing.Load() < mcount()-int32(extraMLength.Load())) && i < 10 {
-				i++
-				usleep(500 * 1000)
+			// Sleep for short intervals so that we can crash quickly after all ms have received SIGQUIT.
+			// Reset the timer whenever we see more ms received SIGQUIT
+			// to make it have enough time to crash (see issue #64752).
+			timeout := watchdogTimeoutMicros
```

Comment:
So many variable assignments I can see, just suggesting a small piece of code below

```
for timeout := watchdogTimeoutMicros; timeout > 0 && crashing.Load() < mcount()-int32(extraMLength.Load()); timeout -= crashSleepMicros {
    usleep(crashSleepMicros)
    if crashing.Load() > maxCrashing {
        maxCrashing = crashing.Load()
        timeout = watchdogTimeoutMicros
    }
}

```

Reply:
Thanks, this pr has pending for Go 1.22 release before since the test failure may caused by another [issue](https://github.com/golang/go/issues/64752#issuecomment-1885242770). So this may not able to fix the issue, see [gerrit](https://go-review.googlesource.com/c/go/+/554615) for more details.